actors/builtin/miner/miner_state.go

```diff
@@ -655,48 +655,10 @@ func (st *State) PopEarlyTerminations(store adt.Store, maxPartitions, maxSectors
 	return result, !noEarlyTerminations, nil
 }
 
-// Returns an error if the target sector cannot be found and/or is faulty/terminated.
-func (st *State) CheckSectorHealth(store adt.Store, dlIdx, pIdx uint64, sector abi.SectorNumber) error {
-	dls, err := st.LoadDeadlines(store)
-	if err != nil {
-		return err
-	}
-
-	dl, err := dls.LoadDeadline(store, dlIdx)
-	if err != nil {
-		return err
-	}
-
-	partition, err := dl.LoadPartition(store, pIdx)
-	if err != nil {
-		return err
-	}
-
-	if exists, err := partition.Sectors.IsSet(uint64(sector)); err != nil {
-		return xc.ErrIllegalState.Wrapf("failed to decode sectors bitfield (deadline %d, partition %d): %w", dlIdx, pIdx, err)
-	} else if !exists {
-		return xc.ErrNotFound.Wrapf("sector %d not a member of partition %d, deadline %d", sector, pIdx, dlIdx)
-	}
-
-	if faulty, err := partition.Faults.IsSet(uint64(sector)); err != nil {
-		return xc.ErrIllegalState.Wrapf("failed to decode faults bitfield (deadline %d, partition %d): %w", dlIdx, pIdx, err)
-	} else if faulty {
-		return xc.ErrForbidden.Wrapf("sector %d of partition %d, deadline %d is faulty", sector, pIdx, dlIdx)
-	}
-
-	if terminated, err := partition.Terminated.IsSet(uint64(sector)); err != nil {
-		return xc.ErrIllegalState.Wrapf("failed to decode terminated bitfield (deadline %d, partition %d): %w", dlIdx, pIdx, err)
-	} else if terminated {
-		return xc.ErrNotFound.Wrapf("sector %d of partition %d, deadline %d is terminated", sector, pIdx, dlIdx)
-	}
-
-	return nil
-}
-
 // Returns an error if the target sector cannot be found, or some other bad state is reached.
 // Returns false if the target sector is faulty, terminated, or unproven
 // Returns true otherwise
-func (st *State) CheckSectorHealthExcludeUnproven(store adt.Store, dlIdx, pIdx uint64, sector abi.SectorNumber) (bool, error) {
+func (st *State) CheckSectorHealthExcludeUnproven(store adt.Store, dlIdx, pIdx uint64, sector abi.SectorNumber, requireProven bool) (bool, error) {
```

Comment:
Why add this if it's never being called with false?

Reply:
So that someone doesn't copy-paste this method in the future when they want a minor change in functionality.

I can remove this flag if @ZenGround0 concurs.

Reply:
LGTM